### PR TITLE
Feat (quant/float): configurable float bit-width implementations

### PR DESCRIPTION
**Pull request by Giuseppe5**
## Risk Highlight

- [ ] This PR includes code from another work (please detail).
- [ ] This PR contains API-breaking changes.
- [ ] This PR depends on work in another PR (please provide links/details).
- [ ] This PR introduces new dependencies (please detail).
- [ ] There are coverage gaps not covered by tests.
- [ ] Documentation updates required in subsequent PR.

## Checklist

- [ ] Code comments added to any hard-to-understand areas, if applicable.
- [ ] Changes generate no new warnings.
- [ ] Updated any relevant tests, if applicable.
- [ ] No conflicts with destination `dev` branch.
- [ ] I reviewed my own code changes.
- [ ] Initial CI/CD passing.
- [ ] 1+ reviews given, and any review issues addressed and approved.
- [ ] Post-review full CI/CD passing.

### CODE REVIEW

File: src/brevitas/function/ops.py [hunk 1]
```diff
@@ -201,8 +201,8 @@ def min_int(
     return value
 
 
-def compute_max_mantissa(mantissa_bit_width: Tensor):
-    return torch.sum((2. ** torch.arange(0, -1. * mantissa_bit_width - 1., -1.)))
+def compute_max_mantissa(mantissa_bit_width: Tensor) -> Tensor:
+    return torch.sum((2. ** torch.arange(0, -1. * mantissa_bit_width.data - 1., -1.)))
```
Comment (Giuseppe5) [inline src/brevitas/function/ops.py:207]: If we don't use `.data`, gradient computation fails

Comment (nickfraser) [reply src/brevitas/function/ops.py:207]: Can you include this as a code comment?

**Review by nickfraser — APPROVED** [review]
Only 1 comment to address, other LGTM!
